js/index.js

```diff
@@ -1,93 +1,141 @@
+        removeButton.innerHTML = "Remove";
+        removeButton.setAttribute("type", "button");
+        removeButton.setAttribute("class", "remove_button");
+
+        removeButton.addEventListener('click', function(){
+            const entry = removeButton.parentNode;
+            entry.removeChild(removeButton);
+            newMessage.remove();
+
+        // To hide the message section when empty
+            if(messageList.children.length === 0) {
+                messageSection.style.display = 'none';
+                leaveMessage.append(footer);
+            }
+        });  
     
-    const username = event.target.usersName.value;
-    const email = event.target.usersEmail.value;
-    const message = event.target.usersMessage.value;
-    
-    console.log(username, email, message);
-
-    const messageSection = document.querySelector('#messages');
-    const messageList = messageSection.querySelector('ul');
-    const newMessage = document.createElement('li'); //makes a new list item
-    messageList.appendChild(newMessage);
-
-    newMessage.innerHTML = `
-    <a href="mailto:${email}">${username}</a>
-    <span> ${message}</span>`;
-
-    // Remove button creation
-    const removeButton = document.createElement('button');
-    removeButton.innerHTML = "Remove";
-    removeButton.setAttribute("type", "button");
-    removeButton.setAttribute("class", "remove_button");
-
-    removeButton.addEventListener('click', function(){
-        const entry = removeButton.parentNode;
-        entry.removeChild(removeButton);
-        newMessage.remove();
-
-    // To hide the message section when empty
-        if(messageList.children.length === 0) {
-            messageSection.style.display = 'none';
-            leaveMessage.append(el);
+    //   Creating the Edit button
+        const editButton = document.createElement('button');
+        editButton.innerHTML = "Edit";
+        editButton.setAttribute("type", "button");
+        editButton.setAttribute("class", "edit_button");
+
+        editButton.addEventListener('click', function(){
+        newMessage.innerHTML = `<a href="mailto:${email}">${username}</a>
+        <textarea> ${message}</textarea>`
+
+        //Creating the second Submit button
+        const resubmitButton = document.createElement('button');
+        resubmitButton.innerHTML = "Submit";
+        resubmitButton.setAttribute("type", "submit");
+        resubmitButton.setAttribute("class", "resubmit_button");
+        newMessage.appendChild(resubmitButton);
+        newMessage.appendChild(removeButton);
+
+        const resubmitForm = document.querySelector('[name="resubmit_message"]');
+
+        resubmitForm.addEventListener('submit', () => {
+            event.preventDefault();
+            // const _message = event.target.querySelector('div').value;
+            // console.log(_message);
+            const contenteditable = document.querySelector('[contenteditable]');
+            const text = contenteditable.innerHTML;
```

Comment:
My suggestion for finding the editable text area in the DOM is to append it to the message like you do with the buttons. You would still create the name and attach it's email by setting the innerHTML of `newMessage`, but you would use `createElement` to make the new textarea. That way you could give it an id and `getElementbyId` instead.

Reply:
I don't think Im following. So I should not try finding 'contenteditable'?